matisse/src/main/java/com/zhihu/matisse/internal/model/SelectedItemCollection.java

```diff
@@ -137,6 +138,16 @@ public void overwrite(ArrayList<Item> items, int collectionType) {
         mItems.addAll(items);
     }
 
+    public void filterInvalidItems() {
+        Iterator<Item> iterator = mItems.iterator();
+        while (iterator.hasNext()) {
+            final Uri uri = iterator.next().getContentUri();
+            if (PathUtils.getPath(mContext, uri) == null) {
+                iterator.remove();
+            }
+        }
+    }
```

Comment:
When users restarted the picker activity, we will reload the album list first, and then check out whether the previously selected items still exist, if users deleted the image, `PathUtils` will resolve nothing, that means we should move it out of selections.